lib/core/string.toit

```diff
@@ -1334,6 +1334,18 @@ abstract class string implements Comparable io.Data:
   write-to-byte-array_ byte-array/ByteArray start end dest-index:
     #primitive.core.string-write-to-byte-array
 
+  byte-size -> int:
+    return size
+
+  byte-slice from to/int -> io.Data:
+    if not 0 <= from <= to <= byte-size: throw "OUT_OF_BOUNDS"
+    if this is String_: return StringByteSlice_ (this as String_) from to
+    slice := this as StringSlice_
+    return StringByteSlice_ slice.str_ (slice.from_ + from) (slice.from_ + to)
+
+  byte-at index/int -> int:
+    return at --raw index
```

Comment:
Since we don't have inlining, perhaps call _raw_at directly instead of going through the indirection.

Reply:
I hope that the `byte-at` function isn't really used, but
done.